test/integration/bad/lang/stanc.expected

```diff
@@ -502,16 +502,16 @@ Syntax error in 'bad_cov_exp_quad_vec_rvec_param.stan', line 17, column 0 to col
 
 "model {" or "generated quantities {" expected after end of transformed parameters block.
   $ ../../../../../install/default/bin/stanc bad_decl1.stan
-Syntax error in 'bad_decl1.stan', line 2, column 14 to column 15, parsing error:
+Syntax error in 'bad_decl1.stan', line 3, column 10 to column 13, parsing error:
    -------------------------------------------------
      1:  data {
-     2:    real x, int y;
-                       ^
-     3:  }
+     2:    // unfortunately the error here got much worse
+     3:    real x, int y;
+                   ^
+     4:  }
    -------------------------------------------------
 
-Expected a new identifier after comma in declaration.
-All variables declared must be of the same type, and any initializing assignment must follow the identifier before the next comma.
+Identifier 'int' clashes with reserved keyword.
```

Comment:
I don't think it got _much_ worse. The old error is pretty bad because it points to the identifier and says "expected a new identifier." At least the new error points to the problematic keyword instead!
I think my above suggested rewording helps here too.

If you want to special-case this error to be even more informative, it might be possible to split the `reserved` error into two cases
```
let reserved id =
  raise (Errors.SyntaxError (Errors.Parsing
          ("Identifier '" ^ id.name ^ "' clashes with reserved keyword.\n",
            id.id_loc)))

let reserved_decl id =
  raise (Errors.SyntaxError (Errors.Parsing
          ("Expected a new identifier after comma in declaration.\nAll variables declared must be of the same type, and any initializing assignment must follow the identifier before the next comma.\n",
            id.id_loc)))
```
and have two rules for `decl_identifier`
```
decl_identifier:
  | id=identifier { id }
  | id=reserved_word { reserved id }

decl_identifier_after_comma:
  | id=identifier { id }
  | id=reserved_word { reserved_decl id }

reserved_word:
  (* Keywords cannot be identifiers but it is nice to
    let them parse as such to provide a better error *)
  | FUNCTIONBLOCK { build_id "functions" $loc }
  | DATABLOCK { build_id "data" $loc }
  | PARAMETERSBLOCK { build_id "parameters" $loc }
  | MODELBLOCK { build_id "model" $loc }
  | RETURN { build_id "return" $loc }
  ...
```

and then use the use the second one here
```
id_and_optional_assignment(rhs):
  | identifier=decl_identifier_after_comma initial_value=optional_assignment(rhs)
    { Ast.{identifier; initial_value} }
```